arborist/user.go

```diff
@@ -210,6 +215,33 @@ func (user *User) createInDb(db *sqlx.DB) *ErrorResponse {
 	return nil
 }
 
+func (user *User) updateInDb(db *sqlx.DB, name *string, email *string) *ErrorResponse {
+	stmt := `
+		UPDATE usr
+		SET
+			name = COALESCE($1, name),
+			email = COALESCE($2, email)
+		WHERE
+			name = $3
+	`
+	result, err := db.Exec(stmt, name, email, user.Name)
+	if err != nil {
+		// this should only fail because the target name was not unique
+		msg := fmt.Sprintf(`failed to update name to "%s": user with this name already exists`, *name)
```

Comment:
what if a user with `name = $3` doesn't exist at all?

Reply:
In the case that `name = $3` doesn't exist at all, the SQL query still executes successfully, but `rowsAffected` below is `0` and a `404` is returned.